crates/configuration/src/version3/mod.rs

```diff
@@ -1020,6 +1054,16 @@ fn convert_operator_kind(
     match operator_kind {
         metadata::OperatorKind::Equal => query_engine_metadata::metadata::OperatorKind::Equal,
         metadata::OperatorKind::In => query_engine_metadata::metadata::OperatorKind::In,
+        metadata::OperatorKind::LessThan => query_engine_metadata::metadata::OperatorKind::LessThan,
```

Comment:
Are we sure these are two separate types? I've seen a few confusing re-exports in this repo, they could be pointing at the same thing.

Reply:
These tyes are different: `metadata:OperatorKind` is local to this specific configuration version, and represents a part of a (de)serializable configuration. There's a copy of that for each configuration version.

`query_engine_metadata::metadata::OperatorKind` belongs to the connector runtime configuration. It may change overtime, and a serialized configuration needs to be able to map to it.

That said, in practice these types are equivalent right now, and could be the same type. But I followed what seemed to be the existing pattern of keeping the two separate.